scripts/python/packages/parthenon_tools/parthenon_tools/movie2d.py

```diff
@@ -35,24 +35,62 @@ def read(filename, nGhost=0):
     return f
 
 
-def plot_dump(xf, yf, q, name, with_mesh=False):
+def plot_dump(
+    xf, yf, q, name, with_mesh=False, block_ids=[], xi=None, yi=None, xe=None, ye=None
+):
+
+    if xe is None:
+        xe = xf
+    if ye is None:
+        ye = yf
+
     fig = plt.figure()
     p = fig.add_subplot(111, aspect=1)
-    qmin = q.min()
-    qmax = q.max()
+    qm = np.ma.masked_where(np.isnan(q), q)
+    qmin = qm.min()
+    qmax = qm.max()
     NumBlocks = q.shape[0]
     for i in range(NumBlocks):
+        # Plot the actual data, should work if parthenon/output*/ghost_zones = true or false
+        # but obviously no ghost data will be shown if ghost_zones = false
         p.pcolormesh(xf[i, :], yf[i, :], q[i, 0, :, :], vmin=qmin, vmax=qmax)
+
+        # Print the block gid in the center of the block
+        if len(block_ids) > 0:
+            p.text(
+                0.5 * (xf[i, 0] + xf[i, -1]),
+                0.5 * (yf[i, 0] + yf[i, -1]),
+                str(block_ids[i]),
+                fontsize=8,
+                color="w",
+                ha="center",
+                va="center",
+            )
```

Comment:
We're not always going to want this. Can this be behind a flag?

Reply:
By default, `block_ids` is an empty array so not passing it acts as a flag.

Reply:
:+1: works for me